app/connectors/ArrivalConnector.scala

```diff
@@ -42,19 +42,13 @@ class ArrivalConnector @Inject()(http: HttpClient, appConfig: AppConfig) extends
     http.POSTString(url, message)(CustomHttpReader, hc = newHeaders, ec = ec)
   }
 
-  def post(message: String, arrivalId: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse] = {
+  def put(message: String, arrivalId: String, headers: Headers)(implicit headerCarrier: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse] = {
     val url = appConfig.traderAtDestinationUrl + arrivalRoute + Utils.urlEncode(arrivalId)
 
-    val newHeaders = hc.withExtraHeaders(requestHeaders(): _*)
-
-    http.POSTString(url, message)(CustomHttpReader, newHeaders, implicitly)
-  }
-
-  def put(message: String, arrivalId: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse] = {
-    val url = appConfig.traderAtDestinationUrl + arrivalRoute + Utils.urlEncode(arrivalId)
-
-    val newHeaders = hc.withExtraHeaders(requestHeaders(): _*)
+    val newHeaders = headerCarrier
+      .copy(authorization = Some(Authorization(headers.get("Authorization").getOrElse(""))))
+      .withExtraHeaders(requestHeaders(): _*)
```

Comment:
Since this manipulation will be everywhere, can you refactor it so that we have a single method will address the headerCarrier, just to tidy up the code. Can you also use the HeaderNames trait so you can have it use HeaderNames.AUTHORIZATION to make it compile safe.